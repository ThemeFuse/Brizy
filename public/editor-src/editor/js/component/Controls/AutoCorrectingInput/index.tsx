import React, { ChangeEvent } from "react";
import _ from "underscore";
import { Props } from "visual/component/Options/Type";
import { WithClassName } from "visual/utils/options/attributes";
import {
  correctNumber,
  formatInputValue,
  preciseAdd,
  preciseSub
} from "./utils";

const DEBOUNCE_WAIT = 1000;

interface IProps extends Props<number>, WithClassName {
  min: number;
  max: number;
  step: number;
  onFocus?: () => void;
  onBlur?: () => void;
  onTextChange?: (e: string | number) => void;
  onMouseEnter?: () => void;
  onMouseLeave?: () => void;
  size?: number;
}

interface IState {
  text: string | number;
  value: number;
  prevPropsValue: number;
}

export default class AutoCorrectingInput extends React.Component<IProps> {
  static defaultProps = {
    className: "",
    value: 0,
    min: 0,
    max: 100,
    step: 1,
    onFocus: _.noop,
    onBlur: _.noop,
    onTextChange: _.noop,
    onChange: _.noop
  };

  state: IState = {
    text: formatInputValue(this.props.value, this.props.step),
    value: this.props.value,
    prevPropsValue: this.props.value
  };

  static getDerivedStateFromProps(props: IProps, state: IState) {
    if (props.value !== state.prevPropsValue) {
      return {
        text: formatInputValue(props.value, props.step),
        value: props.value,
        prevPropsValue: props.value
      };
    }

    return null;
  }

  componentWillUnmount() {
    this.debouncedOnChange.cancel();
  }

  increment() {
    this.setState(
      (state: IState, props) => {
        const { min, max, step } = props;
        const { text, value } = state;
        const textValue = Number(text);
        const correctedAndIncrementedTextValue = correctNumber(
          preciseAdd(textValue, step),
          min,
          max,
          step
        );

        if (correctedAndIncrementedTextValue !== value) {
          return {
            text: formatInputValue(correctedAndIncrementedTextValue, step),
            value: correctedAndIncrementedTextValue
          };
        }

        return null;
      },
      () => {
        if(this.props.onTextChange) this.props.onTextChange(this.state.text);
        this.props.onChange(this.state.value);
      }
    );
  }

  decrement() {
    this.setState(
      (state: IState, props) => {
        const { min, max, step } = props;
        const { text, value } = state;
        const textValue = Number(text);
        const correctedAndDecrementedTextValue = correctNumber(
          preciseSub(textValue, step),
          min,
          max,
          step
        );

        if (correctedAndDecrementedTextValue !== value) {
          return {
            text: formatInputValue(correctedAndDecrementedTextValue, step),
            value: correctedAndDecrementedTextValue
          };
        }

        return null;
      },
      () => {
        if(this.props.onTextChange) this.props.onTextChange(this.state.text);
        this.props.onChange(this.state.value);
      }
    );
  }

  handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    this.setState({ text: e.target.value }, () => {
      if(this.props.onTextChange) this.props.onTextChange(this.state.text);
      this.debouncedOnChange();
    });
  };

  debouncedOnChange = _.debounce(() => {
    this.setState(
      (state: IState, props) => {
        const { min, max, step } = props;
        const { text, value } = state;
        const textValue = parseFloat(String(text));
        const correctedTextValue = correctNumber(
          !Number.isNaN(textValue)
            ? textValue
            : value,
          min,
          max,
          step
        );

        return {
          text: formatInputValue(correctedTextValue, step),
          value: correctedTextValue
        };
      },
      () => {
        if(this.props.onTextChange) this.props.onTextChange(this.state.text);
        this.props.onChange(this.state.value);
      }
    );
  }, DEBOUNCE_WAIT);

  getText() {
    return this.state.text;
  }

  render() {
    const {
      className,
      min,
      max,
      step,
      onFocus,
      onBlur,
      onMouseEnter,
      onMouseLeave,
      size
    } = this.props;
    const { text } = this.state;

    return (
      <input
        className={className}
        type="number"
        value={text}
        min={min}
        max={max}
        step={step}
        size={size}
        onFocus={onFocus}
        onBlur={onBlur}
        onChange={this.handleChange}
        onMouseEnter={onMouseEnter}
        onMouseLeave={onMouseLeave}
      />
    );
  }
}
