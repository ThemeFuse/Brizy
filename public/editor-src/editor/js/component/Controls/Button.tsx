import React, { ReactElement } from "react";
import classNames from "classnames";
import { WithClassName } from "visual/utils/options/attributes";
import EditorIcon from "visual/component/EditorIcon";

export interface Props extends WithClassName {
  onClick: VoidFunction;
  reverse?: boolean;
  label?: string;
  icon?: string;
  align?: "left" | "center" | "right";
}

export const Button = ({
  className,
  onClick,
  icon,
  label,
  reverse,
  align = "center"
}: Props): ReactElement | null => {
  return label || icon ? (
    <div
      className={classNames(
        "brz-ed-control__button",
        `brz-ed-control__button-${align}`,
        className,
        {
          reverse: !!reverse
        }
      )}
      onClick={onClick}
    >
      {icon && <EditorIcon icon={icon} />}
      <span className={"brz-ed-control__label"}>{label}</span>
    </div>
  ) : null;
};
