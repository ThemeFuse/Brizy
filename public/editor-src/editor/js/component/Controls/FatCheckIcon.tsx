import classNames from "classnames";
import React, { MouseEventHandler, ReactElement, useCallback } from "react";
import { CheckMark } from "visual/component/CheckMark";
import { EditorIcon } from "visual/component/EditorIcon";
import { WithClassName } from "visual/utils/options/attributes";

export interface Props extends WithClassName {
  icon: string;
  label: string;
  onClick?: VoidFunction;
  onCheck?: VoidFunction;
  active?: boolean;
  checked?: boolean;
}

export const FatCheckIcon = ({
  active,
  checked,
  className,
  icon,
  label,
  onCheck,
  onClick
}: Props): ReactElement => {
  const _onCheck = useCallback<MouseEventHandler>(
    (e) => {
      onCheck && e.stopPropagation();
      onCheck?.();
    },
    [onCheck]
  );

  const activeLabel = active ? "text-brand-primary" : "text-white";
  const activeIcon = active ? "!text-brand-primary" : "!text-white";

  const activeBorder = active
    ? "border-brand-primary"
    : "border-brand-options hover:border-options-border-hover";

  return (
    <div
      className={classNames(
        className,
        "brz-ed--fat-icon text-center w-[60px] cursor-pointer"
      )}
      onClick={onClick}
      title={label}
    >
      <div
        className={`brz-ed--fat-icon__wrapper relative border-2 border-solid rounded-[3px] flex items-center justify-center w-[60px] h-[60px] transition-[border-color] duration-200 ease-linear	delay-[0s] ${activeBorder}`}
      >
        <CheckMark
          checked={!!checked}
          onClick={_onCheck}
          className={
            "brz-ed--fat-check-icon__check absolute -top-[5px] -left-[5px]"
          }
        />
        <EditorIcon
          icon={icon}
          className={`text-[16px] transition-[color] duration-200 ease-linear ${activeIcon}`}
        />
      </div>
      <div
        className={`brz-ed--fat-icon__label transition-[color] duration-200 ease-linear text-[12px] leading-[2.4em] overflow-hidden text-ellipsis whitespace-nowrap ${activeLabel}`}
      >
        {label}
      </div>
    </div>
  );
};
