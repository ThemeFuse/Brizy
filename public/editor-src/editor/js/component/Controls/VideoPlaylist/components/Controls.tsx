import React from "react";
import { ThemeIcon } from "visual/component/ThemeIcon";
import { FCC } from "visual/utils/react/types";
import { ControlProps as Props } from "../types";

export const Controls: FCC<Props> = ({ hasPauseIcon, hasUnmuteIcon }) => (
  <>
    <div className="brz-video-custom-play-pause-btn">
      <ThemeIcon
        className="brz-icon-svg brz-video-custom-play"
        name="button-play"
        type="glyph"
      />
      {hasPauseIcon && (
        <ThemeIcon
          className="brz-hidden brz-icon-svg brz-video-custom-pause"
          name="button-pause"
          type="glyph"
        />
      )}
    </div>
    <div className="brz-video-custom-controls">
      <span className="brz-video-custom-current-time">0:00</span>

      <div className="brz-video-custom-slider">
        <div className="brz-video-custom-progress" />
      </div>

      <span className="brz-video-custom-total-time">0:00</span>
    </div>
    <div className="brz-video-custom-volume">
      <div className="brz-video-custom-fullscreen-btn">
        <ThemeIcon
          className="brz-icon-svg brz-video-custom-fullscreen-icon"
          name="zoom-e"
          type="glyph"
        />
      </div>
      <div className="brz-video-custom-volume-btn">
        <ThemeIcon
          className="brz-icon-svg brz-video-custom-mute"
          name="volume-97"
          type="glyph"
        />
        {hasUnmuteIcon && (
          <ThemeIcon
            className="brz-hidden brz-icon-svg brz-video-custom-unmute"
            name="volume-ban"
            type="glyph"
          />
        )}
      </div>
      <div className="brz-video-custom-volume-controls brz-video-custom-slider">
        <div className="brz-video-custom-progress" />
      </div>
    </div>
  </>
);
