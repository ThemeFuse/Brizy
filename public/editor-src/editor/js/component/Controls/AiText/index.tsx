import { EnterRounded } from "@brizy/ui-icons";
import { Spacing } from "@brizy/ui/es/Space/utils";
import { FieldsTheme } from "@brizy/ui/es/utils/getFieldsTheme";
import { Gutter } from "@brizy/ui/lib/Layout/utils";
import React, { ReactElement, useCallback, useEffect, useState } from "react";
import { Button } from "visual/component/Brizy-ui/Button";
import { Card } from "visual/component/Brizy-ui/Card";
import { Icon } from "visual/component/Brizy-ui/Icon";
import { Inline } from "visual/component/Brizy-ui/Inline";
import { Input } from "visual/component/Brizy-ui/Input";
import { Layout, LayoutSection } from "visual/component/Brizy-ui/Layout";
import { Space } from "visual/component/Brizy-ui/Space";
import { t } from "visual/utils/i18n";
import { cardSize } from "./constants";
import { AiActionButton } from "./controls/AiActionButton";
// import { AiDropdown } from "./controls/AiDropdown";
// import { data } from "./data/aiText";
import { actions } from "./data/aiTextActions";
// import { DropdownButtonData } from "./types";

const theme: FieldsTheme = {
  background: "dark",
  borderColor: "dark",
  color: "white"
};

const controlsSpacing: Spacing = [0, 0, 15, 10];
const textareaSpacing: Spacing = [0, 8, 5, 0];
const textareaButtonSpacing: Spacing = [0, 0, 0, 5];
const promptAreaGutter: Gutter = [0, 0];

interface Props {
  onChange: (v: string) => void;
  value: string;
  loading: boolean;
  submitRequest: (prompt: string, action?: string) => void;
  prompt: string;
  isTitleSelected: boolean;
}

export const AiText = ({
  onChange,
  value,
  loading,
  submitRequest,
  prompt,
  isTitleSelected
}: Props): ReactElement => {
  const [disabled, setDisabled] = useState<boolean>(false);
  const [buttonDisabled, setButtonDisabled] = useState<boolean>(true);

  const iconColor = value.length > 0 ? "white" : "gray";

  const handleClick = useCallback(
    (action: string) => {
      submitRequest(prompt, action);
    },
    [prompt, submitRequest]
  );

  const handlePressEnter = useCallback(() => {
    if (!loading && value.length > 0) {
      submitRequest(value);
    }
  }, [loading, value, submitRequest]);

  useEffect(() => {
    if (prompt.length <= 3) {
      setDisabled(true);
    }
  }, [prompt]);

  useEffect(() => {
    setButtonDisabled(value.length === 0 || loading);
  }, [value, loading]);

  return (
    <Card width="407px" size={cardSize} borderStyle="none" color="transparent">
      <Space spacing={controlsSpacing}>
        <Card size={cardSize} width="407px" height="16px">
          <Inline spacing={20}>
            {actions.map((action, index) => (
              <AiActionButton
                isTitleSelected={isTitleSelected}
                disabled={disabled}
                key={index}
                onClick={handleClick}
                action={action}
              />
            ))}
            {/* temporary commented https://github.com/bagrinsergiu/blox-editor/issues/26546*/}
            {/* {data.map((item: DropdownButtonData, index: number) => (
              <AiDropdown
                disabled={disabled}
                onOptionClick={handleClick}
                data={item}
                key={index}
              />
            ))} */}
          </Inline>
        </Card>
      </Space>
      <Space spacing={textareaSpacing}>
        <Layout alignY="middle" gutter={promptAreaGutter}>
          <LayoutSection span={23}>
            <Input.TextArea
              theme={theme}
              placeholder={t("Tell AI what to write in any language ...")}
              onChange={onChange}
              value={value}
              size="xsmall"
              onPressEnter={handlePressEnter}
            />
          </LayoutSection>
          <LayoutSection span={1}>
            <Space spacing={textareaButtonSpacing}>
              <Button
                color="white"
                hoverColor="white"
                type="dashed"
                loading={loading}
                disabled={buttonDisabled}
                icon={
                  <Icon size="20px" source={EnterRounded} color={iconColor} />
                }
                onClick={handlePressEnter}
              ></Button>
            </Space>
          </LayoutSection>
        </Layout>
      </Space>
    </Card>
  );
};
