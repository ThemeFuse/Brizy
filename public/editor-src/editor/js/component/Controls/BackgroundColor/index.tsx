import React, { FC } from "react";
import classNames from "classnames";
import { t } from "visual/utils/i18n";
import { WithClassName, WithValue } from "visual/utils/options/attributes";
import AutoCorrectingInput from "visual/component/Controls/AutoCorrectingInput";
import {
  ColorPickerSelect,
  Props as CSProps
} from "visual/component/Controls/ColorPickerSelect";
import { Item } from "visual/component/Controls/MultiSelect/Item";
import { PaletteObject } from "visual/component/Controls/ColorPalette/entities/PaletteObject";
import { Value, Meta, Type, GradientType } from "./entities";
import { fromColorMeta } from "./utils";
import { ColorPickerInputs } from "visual/component/Controls/ColorPicketInputs";
import { Select2 } from "visual/component/Controls/Select2";
import { GradientRange } from "./GradientRange";

type Props = WithClassName &
  WithValue<Value> & {
    onChange: (v: Value, m: Meta) => void;
    paletteOpenSettings: () => void;
    palette: PaletteObject[];
    opacity: boolean;
    gradientColors: [string, string];
  };

export const BackgroundColor: FC<Props> = ({
  className,
  onChange,
  value,
  paletteOpenSettings,
  palette,
  opacity,
  gradientColors: [g1, g2]
}) => {
  const onColorChange: CSProps<Type>["onChange"] = (
    (v, m) => {
      onChange(
        {
          type: v.select,
          hex: v.hex,
          opacity: v.opacity,
          palette: v.palette,
          gradientType: value.gradientType,
          start: value.start,
          end: value.end,
          active: value.active,
          degree: value.degree
        },
        fromColorMeta(m)
      );
    }
  );
  const onSelectChange = (
    (gradientType: GradientType): void =>
      onChange({ ...value, gradientType }, { isChanged: "gradientType" })
  );
  const _className = classNames("brz-ed-control__backgroundColor", className);
  return (
    <div className={_className}>
      {value.type === "gradient" ? (
        <GradientRange
          start={value.start}
          end={value.end}
          active={value.active}
          hex1={g1}
          hex2={g2}
          onStartChange={(start): void =>
            onChange({ ...value, start }, { isChanged: "start" })
          }
          onEndChange={(end): void =>
            onChange({ ...value, end }, { isChanged: "end" })
          }
          onActiveChange={(active): void =>
            onChange({ ...value, active }, { isChanged: "active" })
          }
        />
      ) : null}
      <ColorPickerSelect<Type>
        palette={palette}
        opacity={opacity}
        value={{
          hex: value.hex,
          palette: value.palette,
          opacity: value.opacity,
          select: value.type
        }}
        onChange={onColorChange}
        paletteOpenSettings={paletteOpenSettings}
      >
        <Item<Type> value={"none"}>{t("None")}</Item>
        <Item<Type> value={"solid"}>{t("Solid")}</Item>
        <Item<Type> value={"gradient"}>{t("Gradient")}</Item>
      </ColorPickerSelect>
      <ColorPickerInputs
        value={value.hex}
        onChange={(hex): void =>
          onChange({ ...value, hex }, { isChanged: "hex" })
        }
      >
        {value.type === "gradient" ? (
          <>
            <Select2<GradientType>
              editable={false}
              className={classNames(
                "brz-ed-control__backgroundColor__gradientType",
                `brz-ed-control__backgroundColor__gradientType--${value.gradientType}`
              )}
              onChange={onSelectChange}
              value={value.gradientType}
            >
              <Item<GradientType> value={"linear"}>{t("Linear")}</Item>
              <Item<GradientType> value={"radial"}>{t("Radial")}</Item>
            </Select2>
            <div className="brz-ed-control__backgroundColor--degree">
              <AutoCorrectingInput
                className="brz-input"
                min={-359}
                max={359}
                step={1}
                value={value.degree}
                onChange={(degree: number): void =>
                  onChange({ ...value, degree }, { isChanged: "degree" })
                }
              />
            </div>
          </>
        ) : null}
      </ColorPickerInputs>
    </div>
  );
};
