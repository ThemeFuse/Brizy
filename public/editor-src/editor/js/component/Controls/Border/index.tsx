import classNames from "classnames";
import React, { FC, useCallback } from "react";
import { PaletteObject } from "visual/component/Controls/ColorPalette/entities/PaletteObject";
import { ColorPickerSelect } from "visual/component/Controls/ColorPickerSelect";
import {
  Meta,
  Value
} from "visual/component/Controls/ColorPickerSelect/entities";
import { ColorPickerInputs } from "visual/component/Controls/ColorPicketInputs";
import { Item } from "visual/component/Controls/MultiSelect/Item";
import EditorIcon from "visual/component/EditorIcon";
import { OnChange } from "visual/component/Options/Type";
import * as BorderStyle from "visual/component/Options/types/dev/Border/entities/style";
import * as BorderWidthType from "visual/component/Options/types/dev/Border/entities/widthType";
import * as Palette from "visual/component/Options/types/dev/ColorPicker/entities/palette";
import { t } from "visual/utils/i18n/t";
import { WithClassName } from "visual/utils/options/attributes";
import { mApply } from "visual/utils/value";
import { Width } from "./Width";
import { toggleType } from "./utils";

type Style = BorderStyle.Style;

export type StyleObject = {
  id: Style;
  title?: string;
  icon?: string;
};

export interface Props extends WithClassName {
  enableOpacity: boolean;
  styles: StyleObject[];
  paletteList: PaletteObject[];
  paletteOpenSettings?: () => void;
  widthTypes: ("grouped" | "ungrouped")[];
  style: Style;
  onChangeStyle: OnChange<Style>;
  hex: string;
  onChangeHex: OnChange<string>;
  opacity: number;
  onChangeOpacity: (v: number, editing: boolean) => void;
  palette: Palette.Palette;
  onChangePalette: OnChange<Palette.Palette>;
  widthType: BorderWidthType.WidthType;
  onChangeWidthType: OnChange<BorderWidthType.WidthType>;
  width: number;
  onChangeWidth: OnChange<number>;
  topWidth: number;
  onChangeTopWidth: OnChange<number>;
  rightWidth: number;
  onChangeRightWidth: OnChange<number>;
  bottomWidth: number;
  onChangeBottomWidth: OnChange<number>;
  leftWidth: number;
  onChangeLeftWidth: OnChange<number>;
}

export const Border: FC<Props> = ({
  className,
  widthTypes,
  styles,
  hex,
  onChangeHex,
  palette,
  onChangePalette,
  opacity,
  onChangeOpacity,
  style,
  onChangeStyle,
  widthType,
  onChangeWidthType,
  width,
  onChangeWidth,
  topWidth,
  onChangeTopWidth,
  rightWidth,
  onChangeRightWidth,
  bottomWidth,
  onChangeBottomWidth,
  leftWidth,
  onChangeLeftWidth,
  ...props
}) => {
  const _className = classNames("brz-ed-control__border", className);
  const onColor = useCallback(
    (v: Value<Style>, meta: Meta) => {
      switch (meta.isChanged) {
        case "select":
          mApply(onChangeStyle, v.select);
          break;
        case "hex":
          onChangeHex(v.hex);
          break;
        case "opacity":
          onChangeOpacity(v.opacity, meta?.opacityDragEnd ?? true);
          break;
        case "palette":
          onChangePalette(v.palette as Palette.Palette);
          break;
      }
    },
    [
      hex,
      onChangeHex,
      palette,
      onChangePalette,
      opacity,
      onChangeOpacity,
      style,
      onChangeStyle
    ]
  );
  const onTypeChange = useCallback(
    () => onChangeWidthType(toggleType(widthType)),
    [widthType, onChangeWidthType]
  );

  const buttonClassName = classNames("brz-button", {
    "brz-ed-control__border__widthType": true,
    "brz-ed-control__border__widthType--active": widthType === "grouped"
  });

  const grouped = widthType === "grouped" || !widthTypes.includes("ungrouped");

  return (
    <div className={_className}>
      <ColorPickerSelect<Style>
        onChange={onColor}
        value={{
          select: style,
          hex: hex,
          opacity: opacity,
          palette: palette
        }}
        opacity={props.enableOpacity}
        palette={props.paletteList}
        paletteOpenSettings={props.paletteOpenSettings}
      >
        {styles.map(({ id, title, icon }) => {
          return (
            <Item<Style> key={id} value={id}>
              {title && <span>{title}</span>}
              {icon && <EditorIcon icon={icon} className="brz-icon" />}
            </Item>
          );
        })}
      </ColorPickerSelect>
      <ColorPickerInputs value={hex} onChange={onChangeHex}>
        {grouped ? (
          <div className="brz-ed-control__border__width">
            <div className="brz-label">{t("Size")}</div>
            <Width value={width} onChange={onChangeWidth} />
          </div>
        ) : (
          <div className="brz-ed-control__border__width">
            <Width value={topWidth} onChange={onChangeTopWidth} />
            <Width value={rightWidth} onChange={onChangeRightWidth} />
            <Width value={bottomWidth} onChange={onChangeBottomWidth} />
            <Width value={leftWidth} onChange={onChangeLeftWidth} />
          </div>
        )}
        {widthTypes.length < 2 ? null : (
          <button className={buttonClassName} onClick={onTypeChange}>
            <EditorIcon icon={"nc-combined-shape"} className="brz-icon" />
          </button>
        )}
      </ColorPickerInputs>
    </div>
  );
};
