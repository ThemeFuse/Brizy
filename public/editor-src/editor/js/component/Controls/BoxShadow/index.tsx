import classNames from "classnames";
import React, { FC, useCallback } from "react";
import {
  Meta,
  TypeObject,
  Value
} from "visual/component/Controls/BoxShadow/types";
import { getModifiedField } from "visual/component/Controls/BoxShadow/utils";
import { ColorPickerSelect } from "visual/component/Controls/ColorPickerSelect";
import {
  Meta as CMeta,
  Value as CValue
} from "visual/component/Controls/ColorPickerSelect/entities";
import { ColorPickerInputs } from "visual/component/Controls/ColorPicketInputs";
import MultiInputOptionType from "visual/component/Controls/MultiInput";
import { Item } from "visual/component/Controls/Select2/Item";
import * as Palette from "visual/component/Options/types/dev/ColorPicker/entities/palette";
import {
  WithClassName,
  WithOnChange2,
  WithValue
} from "visual/utils/options/attributes";
import { PaletteObject } from "../ColorPalette/entities/PaletteObject";

export interface Props
  extends WithClassName,
    WithValue<Value>,
    WithOnChange2<Value, Meta> {
  opacity: boolean;
  types: TypeObject[];
  palette: PaletteObject[];
  paletteOpenSettings?: () => void;
}

export const BoxShadow: FC<Props> = ({ value, types, onChange, ...props }) => {
  const className = classNames("brz-ed-control__boxShadow", props.className);
  const onHex = useCallback(
    (hex) => onChange({ ...value, hex }, { isChanged: "hex" }),
    [onChange, value]
  );
  const fieldChange = useCallback(
    (fields) => {
      const [blur, spread, vertical, horizontal] = fields;
      const current = [
        value.blur,
        value.spread,
        value.vertical,
        value.horizontal
      ];
      const isChanged = getModifiedField(fields, current);
      if (isChanged) {
        onChange(
          { ...value, blur, spread, vertical, horizontal },
          { isChanged }
        );
      }
    },
    [onChange, value]
  );
  const _onChange = (
    { select, ...v }: CValue<Value["type"]>,
    m: CMeta
  ): void => {
    onChange(
      {
        ...value,
        hex: v.hex,
        opacity: v.opacity,
        palette: v.palette as Palette.Palette,
        type: select
      },
      { ...m, isChanged: m.isChanged === "select" ? "type" : m.isChanged }
    );
  };

  return (
    <div className={className}>
      <ColorPickerSelect<Value["type"]>
        onChange={_onChange}
        value={{
          select: value.type,
          hex: value.hex,
          opacity: value.opacity,
          palette: value.palette
        }}
        opacity={props.opacity}
        palette={props.palette}
        paletteOpenSettings={props.paletteOpenSettings}
      >
        {types.map(({ id, title }) => {
          return (
            <Item<Value["type"]> value={id} key={id}>
              <span>{title}</span>
            </Item>
          );
        })}
      </ColorPickerSelect>
      <ColorPickerInputs value={value.hex} onChange={onHex}>
        <MultiInputOptionType
          config={{
            defaultIcon: "nc-shadow",
            icons: ["nc-blur", "nc-size", "nc-vertical", "nc-horizontal"]
          }}
          value={[value.blur, value.spread, value.vertical, value.horizontal]}
          onChange={fieldChange}
        />
      </ColorPickerInputs>
    </div>
  );
};
