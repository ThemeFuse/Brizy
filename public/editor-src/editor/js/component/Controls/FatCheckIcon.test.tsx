import React from "react";
import { render } from "@testing-library/react";
import { FatCheckIcon } from "visual/component/Controls/FatCheckIcon";
import Config from "visual/global/Config";

describe("Snapshots", () => {
  beforeAll(() => {
    global.IS_EDITOR = true;

    Config.load({
      // @ts-expect-error to find out what happens here
      urls: {
        editorIcons: "/"
      }
    });
  });

  test("Default view", () => {
    const node = render(<FatCheckIcon icon={"nc-button"} label={"Test"} />);

    expect(node.container.firstChild).toMatchSnapshot();
  });
});

describe("Test props", () => {
  test("Put label text on container element as title attribute", () => {
    const node = render(
      <FatCheckIcon icon={"nc-button"} label={"Test label"} />
    );

    expect(node.container.firstElementChild?.getAttribute("title")).toBe(
      "Test label"
    );
  });
});
