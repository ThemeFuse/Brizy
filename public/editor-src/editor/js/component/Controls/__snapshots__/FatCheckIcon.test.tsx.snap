// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots Default view 1`] = `
<div
  class="brz-ed--fat-icon text-center w-[60px] cursor-pointer"
  title="Test"
>
  <div
    class="brz-ed--fat-icon__wrapper relative border-2 border-solid rounded-[3px] flex items-center justify-center w-[60px] h-[60px] transition-[border-color] duration-200 ease-linear	delay-[0s] border-right-sidebar-item-border hover:border-right-sidebar-item-border--hover"
  >
    <span
      class="brz-ed--check-mark w-[16px] h-[16px] text-[16px] rounded-[50%] top-[4px] left-[8px] bg-black border-solid border-topaz brz-ed--fat-check-icon__check absolute -top-[5px] -left-[5px] border-[1px]"
    >
      <svg
        class="brz-icon-svg brz-ed-icon-svg align-[initial] !text-brand-primary opacity-0"
      >
        <use
          xlink:href="//icons.svg#nc-check-circle"
        />
      </svg>
    </span>
    <svg
      class="brz-icon-svg brz-ed-icon-svg align-[initial] text-[16px] transition-[color] duration-200 ease-linear !text-right-sidebar-item"
    >
      <use
        xlink:href="//icons.svg#nc-button"
      />
    </svg>
  </div>
  <div
    class="brz-ed--fat-icon__label transition-[color] duration-200 ease-linear text-[12px] leading-[2.4em] overflow-hidden text-ellipsis whitespace-nowrap text-right-sidebar-item"
  >
    Test
  </div>
</div>
`;
